hexl/fft/fft-native.cpp

```diff
@@ -0,0 +1,414 @@
+// Copyright (C) 2020-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hexl/fft/fft-native.hpp"
+
+#include <cstring>
+
+#include "util/util-internal.hpp"
+
+namespace intel {
+namespace hexl {
+
+inline void ComplexFwdButterflyRadix2(std::complex<double_t>* X_r,
+                                      std::complex<double_t>* Y_r,
+                                      const std::complex<double_t>* X_op,
+                                      const std::complex<double_t>* Y_op,
+                                      const std::complex<double_t> W) {
+  std::complex<double_t> U = *X_op;
+  std::complex<double_t> V = *Y_op * W;
+  *X_r = U + V;
+  *Y_r = U - V;
+}
+
+inline void ComplexInvButterflyRadix2(std::complex<double_t>* X_r,
+                                      std::complex<double_t>* Y_r,
+                                      const std::complex<double_t>* X_op,
+                                      const std::complex<double_t>* Y_op,
+                                      const std::complex<double_t> W) {
+  std::complex<double_t> U = *X_op;
+  *X_r = U + *Y_op;
+  *Y_r = U - *Y_op;
+  *Y_r *= W;
+}
+
+inline void ScaledComplexInvButterflyRadix2(std::complex<double_t>* X_r,
+                                            std::complex<double_t>* Y_r,
+                                            const std::complex<double_t>* X_op,
+                                            const std::complex<double_t>* Y_op,
+                                            const std::complex<double_t> W,
+                                            const double_t* scalar) {
+  std::complex<double_t> U = *X_op;
+  *X_r = U + *Y_op;
+  *X_r *= *scalar;
+  *Y_r = U - *Y_op;
+  *Y_r *= W;
+}
+
+void Forward_FFT_ToBitReverseRadix2(
+    std::complex<double_t>* result, const std::complex<double_t>* operand,
```

Comment:
I see many double_t instead of just double. They don't quite mean the same thing although the compiler may treat them as such depending on impl. Was this what you wanted? and why?

Reply:
No real reason, I will change to doubles to avoid any confusion.